test_common/page_objects/base_page.rb

```diff
@@ -15,6 +15,17 @@ def self.set_url(url)
         element :more_category_link, :link_named, 'components.sidebar.more_category_link'
       end
 
+      def js_severe_errors
+        return [] unless page.driver.try(:browser).try(:browser) == :chromedriver
+        errors = page.driver.browser.manage.logs.get(:browser).select { |error| error.level == "SEVERE" }
+        errors.map do |error|
+          { message: error.message }
+        end
+      end
+
+      def clear_browser_logs
+        page.driver.browser.manage.logs.get(:browser).delete_if { |x| x }
```

Comment:
I dont like this - I think it will give you problems.  But, I cannot be 100% sure.

Effectively, you are doing a 'delete all' on every element of the array returned by page.driver.browser.manage.logs.get(:browser) - which will not delete the logs in the browser itself.  Instead, it will mean that this array will not be in touch with reality anymore, causing false positives or negatives in your tests.

Reply:
Removed in f5adb6d